service/src/test/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderServiceTest.java

```diff
@@ -0,0 +1,174 @@
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HIS_HONOUR_JUDGE;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createPopulatedChildren;
+
+@ExtendWith(SpringExtension.class)
+class CreateC21OrderServiceTest {
+    private static final String LOCAL_AUTHORITY_CODE = "example";
+    private static final String COURT_NAME = "Example Court";
+    private static final String COURT_EMAIL = "example@court.com";
+    private static final String CONFIG = String.format("%s=>%s:%s", LOCAL_AUTHORITY_CODE, COURT_NAME, COURT_EMAIL);
+    private static final LocalDateTime NOW = LocalDateTime.now();
+
+    private final Time time = () -> NOW;
+
+    private DateFormatterService dateFormatterService = new DateFormatterService();
+    private HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration = new HmctsCourtLookupConfiguration(CONFIG);
+    private CreateC21OrderService createC21OrderService = new CreateC21OrderService(dateFormatterService,
+        hmctsCourtLookupConfiguration, time);
+
+    @Test
+    void shouldAppendNewC21OrderToEmptyC21OrderBundle() {
+        List<Element<C21OrderBundle>> emptyC21OrderBundle = new ArrayList<>();
+        CaseData caseData = addC21OrderAndBundleToCaseData(emptyC21OrderBundle, "C21_1.pdf");
+
+        List<Element<C21OrderBundle>> c21OrderBundleWithOrder = createC21OrderService.addToC21OrderBundle(
+            caseData.getTemporaryC21Order(), caseData.getJudgeAndLegalAdvisor(), caseData.getC21OrderBundle());
+        assertThat(c21OrderBundleWithOrder.size()).isEqualTo(1);
+
+        C21OrderBundle c21OrderBundle = c21OrderBundleWithOrder.get(0).getValue();
+
+        assertThat(c21OrderBundle.getC21OrderDocument().getFilename()).isEqualTo("C21_1.pdf");
+        assertThat(c21OrderBundle.getOrderDate()).isEqualTo(
+            dateFormatterService.formatLocalDateTimeBaseUsingFormat(time.now(), "h:mma, d MMMM yyyy"));
+        assertThat(c21OrderBundle.getOrderTitle()).isEqualTo("Example order title");
+        assertThat(c21OrderBundle.getJudgeTitleAndName()).isEqualTo("His Honour Judge Johnson");
+    }
+
+    @Test
+    void shouldAppendNewC21OrderToExistingC21OrderBundle() {
+        CaseData caseData = addC21OrderAndBundleToCaseData(getExistingC21OrderBundle(), "C21_2.pdf");
+
+        List<Element<C21OrderBundle>> c21OrderBundleWithTwoOrders = createC21OrderService.addToC21OrderBundle(
+            caseData.getTemporaryC21Order(), caseData.getJudgeAndLegalAdvisor(), caseData.getC21OrderBundle());
+        assertThat(c21OrderBundleWithTwoOrders.size()).isEqualTo(2);
+        C21OrderBundle previousC21 = c21OrderBundleWithTwoOrders.get(0).getValue();
+        C21OrderBundle appendedC21 = c21OrderBundleWithTwoOrders.get(1).getValue();
+
+        assertThat(previousC21.getC21OrderDocument().getFilename()).isEqualTo("C21_1.pdf");
+        assertThat(appendedC21.getC21OrderDocument().getFilename()).isEqualTo("C21_2.pdf");
+    }
```

Comment:
ok so again let's not bother with adding to caseData just to remove again. Please just build the objects and use them in the method. If you do this in a private method, this would probably be best. Please space this out a little better... Keep assertions bundled together and use .hasSize() in your assertion